### ISSUE TITLE
sourceCode.isSpaceBetweenTokens() and multiline comments

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#12263]
Noticed this while working on #12259 

`sourceCode.isSpaceBetweenTokens()` sees newlines inside multiline block comments as spaces.

Observable:

```js
/* eslint space-before-function-paren:["error", "never"]*/

function foo/*
*/(){}
```

[Demo link](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHNwYWNlLWJlZm9yZS1mdW5jdGlvbi1wYXJlbjpbXCJlcnJvclwiLCBcIm5ldmVyXCJdKi9cblxuZnVuY3Rpb24gZm9vLypcbiovKCl7fVxuXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjUsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOjIsImZvci1kaXJlY3Rpb24iOjIsImdldHRlci1yZXR1cm4iOjIsIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOjIsIm5vLWNhc2UtZGVjbGFyYXRpb25zIjoyLCJuby1jbGFzcy1hc3NpZ24iOjIsIm5vLWNvbXBhcmUtbmVnLXplcm8iOjIsIm5vLWNvbmQtYXNzaWduIjoyLCJuby1jb25zdC1hc3NpZ24iOjIsIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6Miwibm8tY29udHJvbC1yZWdleCI6Miwibm8tZGVidWdnZXIiOjIsIm5vLWRlbGV0ZS12YXIiOjIsIm5vLWR1cGUtYXJncyI6Miwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjoyLCJuby1kdXBlLWtleXMiOjIsIm5vLWR1cGxpY2F0ZS1jYXNlIjoyLCJuby1lbXB0eSI6Miwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1lbXB0eS1wYXR0ZXJuIjoyLCJuby1leC1hc3NpZ24iOjIsIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6Miwibm8tZXh0cmEtc2VtaSI6Miwibm8tZmFsbHRocm91Z2giOjIsIm5vLWZ1bmMtYXNzaWduIjoyLCJuby1nbG9iYWwtYXNzaWduIjoyLCJuby1pbm5lci1kZWNsYXJhdGlvbnMiOjIsIm5vLWludmFsaWQtcmVnZXhwIjoyLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6Miwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLW1peGVkLXNwYWNlcy1hbmQtdGFicyI6Miwibm8tbmV3LXN5bWJvbCI6Miwibm8tb2JqLWNhbGxzIjoyLCJuby1vY3RhbCI6Miwibm8tcHJvdG90eXBlLWJ1aWx0aW5zIjoyLCJuby1yZWRlY2xhcmUiOjIsIm5vLXJlZ2V4LXNwYWNlcyI6Miwibm8tc2VsZi1hc3NpZ24iOjIsIm5vLXNoYWRvdy1yZXN0cmljdGVkLW5hbWVzIjoyLCJuby1zcGFyc2UtYXJyYXlzIjoyLCJuby10aGlzLWJlZm9yZS1zdXBlciI6Miwibm8tdW5kZWYiOjIsIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjoyLCJuby11bnJlYWNoYWJsZSI6Miwibm8tdW5zYWZlLWZpbmFsbHkiOjIsIm5vLXVuc2FmZS1uZWdhdGlvbiI6Miwibm8tdW51c2VkLWxhYmVscyI6Miwibm8tdXNlbGVzcy1jYXRjaCI6Miwibm8tdXNlbGVzcy1lc2NhcGUiOjIsIm5vLXdpdGgiOjIsInJlcXVpcmUtYXRvbWljLXVwZGF0ZXMiOjIsInJlcXVpcmUteWllbGQiOjIsInVzZS1pc25hbiI6MiwidmFsaWQtdHlwZW9mIjoyfSwiZW52Ijp7fX19)

The above is an error because of the newline (`function foo/* */(){}` is not an error for the same rule.)

Is it intentional or a bug?

There is no such test for `isSpaceBetweenTokens()` in `source-code` tests.

Also, when I add the `s` flag to the `/\/\*.*?\*\//gu` regex in `isSpaceBetweenTokens` and run `npm test` everything still works.

**Comment by kaicataldo** [comment]
I will try to work on this this weekend!

**Comment by kaicataldo** [comment]
@mdjermanovic Though if you need to fix it for your work, please feel free to take it on. I don’t want to block you!

**Comment by mdjermanovic** [comment]
@kaicataldo Feel free to claim this :)

The effects of this change can be also observed in the current version of `template-curly-spacing`, it will have more and fewer errors in some cases.

**Comment by kaicataldo** [comment]
Sorry I haven't gotten to this yet. It's in my backlog, but if someone can get to it before I can, please feel free!

**Comment by mdjermanovic** [comment]
Fixed in #12407
